contrib/aws/aws-sdk-go/aws/aws.go

```diff
@@ -147,3 +163,224 @@ func awsAgent(req *request.Request) string {
+		tableName = *input.TableName
+	case *dynamodb.ScanInput:
+		tableName = *input.TableName
+	case *dynamodb.UpdateItemInput:
+		tableName = *input.TableName
+	default:
+		return nil, nil
+	}
+	return map[string]string{
+		tags.DynamoDBTableName: tableName,
+	}, nil
+}
+
+func kinesisTags(params interface{}) (map[string]string, error) {
+	var streamName string
+	switch input := params.(type) {
+	case *kinesis.PutRecordInput:
+		streamName = *input.StreamName
+	case *kinesis.PutRecordsInput:
+		streamName = *input.StreamName
+	case *kinesis.AddTagsToStreamInput:
+		streamName = *input.StreamName
+	case *kinesis.RemoveTagsFromStreamInput:
+		streamName = *input.StreamName
+	case *kinesis.CreateStreamInput:
+		streamName = *input.StreamName
+	case *kinesis.DeleteStreamInput:
+		streamName = *input.StreamName
+	case *kinesis.DescribeStreamInput:
+		streamName = *input.StreamName
+	case *kinesis.DescribeStreamSummaryInput:
+		streamName = *input.StreamName
+	case *kinesis.GetShardIteratorInput:
+		streamName = *input.StreamName
+	default:
+		return nil, nil
+	}
+	return map[string]string{
+		tags.KinesisStreamName: streamName,
+	}, nil
+}
+
+func eventBridgeTags(params interface{}) (map[string]string, error) {
+	var ruleName string
+	switch input := params.(type) {
+	case *eventbridge.PutRuleInput:
+		ruleName = *input.Name
+	case *eventbridge.DescribeRuleInput:
+		ruleName = *input.Name
+	case *eventbridge.DeleteRuleInput:
+		ruleName = *input.Name
+	case *eventbridge.DisableRuleInput:
+		ruleName = *input.Name
+	case *eventbridge.EnableRuleInput:
+		ruleName = *input.Name
+	case *eventbridge.PutTargetsInput:
+		ruleName = *input.Rule
+	case *eventbridge.RemoveTargetsInput:
+		ruleName = *input.Rule
+	default:
+		return nil, nil
+	}
+	return map[string]string{
+		tags.EventBridgeRuleName: ruleName,
+	}, nil
+}
+
+func sfnTags(params interface{}) (map[string]string, error) {
+	nameFromExecutionARN := func(arn string) string {
+		parts := strings.Split(arn, ":")
+		return parts[len(parts)-2]
+	}
+	var stateMachineName, stateMachineArn string
+	switch input := params.(type) {
+	case *sfn.CreateStateMachineInput:
+		stateMachineName = *input.Name
+	case *sfn.DescribeStateMachineInput:
+		stateMachineArn = *input.StateMachineArn
+	case *sfn.StartExecutionInput:
+		stateMachineArn = *input.StateMachineArn
+	case *sfn.StopExecutionInput:
+		if input.ExecutionArn != nil {
+			stateMachineName = nameFromExecutionARN(*input.ExecutionArn)
+		}
+	case *sfn.DescribeExecutionInput:
+		if input.ExecutionArn != nil {
+			stateMachineName = nameFromExecutionARN(*input.ExecutionArn)
+		}
+	case *sfn.ListExecutionsInput:
+		stateMachineArn = *input.StateMachineArn
+	case *sfn.UpdateStateMachineInput:
+		stateMachineArn = *input.StateMachineArn
+	case *sfn.DeleteStateMachineInput:
+		stateMachineArn = *input.StateMachineArn
+	}
+	if stateMachineName == "" {
+		parts := strings.Split(stateMachineArn, ":")
+		stateMachineName = parts[len(parts)-1]
```

Comment:
Confused by why this one does `parts[len(parts)-1]` but above in line 354 we do `parts[len(parts)-2]`?

Reply:
A state machine ARN looks like this ([ref](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-stepfunctions-statemachine.html)):
`arn:aws:states:us-east-1:111122223333:stateMachine:HelloWorld-StateMachine`

A execution ARN looks like this ([ref](https://docs.aws.amazon.com/step-functions/latest/dg/input-output-contextobject.html)):
`"arn:aws:states:us-east-1:123456789012:execution:stateMachineName:executionName"`

In the first case, the state machine name is the last field, and in the second case, it's the previous to the last one.

I'll add some comments to clarify this!